cranelift/codegen/src/isa/aarch64/inst/mod.rs

```diff
@@ -2491,6 +2496,21 @@ impl MachInst for Inst {
         }
     }
 
+    fn is_included_in_clobbers(&self) -> bool {
+        // We exclude call instructions from the clobber-set when they are calls
+        // from caller to callee with the same ABI. Such calls cannot possibly
+        // force any new registers to be saved in the prologue, because anything
+        // that the callee clobbers, the caller is also allowed to clobber. This
```

Comment:
This rule about caller- vs callee- saved is actually generic over the targets, so we could also apply it for x64, right?

Reply:
Indeed it is, although we get less benefit (only the compile-time improvement, no generated-code change) because x64 doesn't have half-caller/half-callee saves. Since it adds overhead to `Inst::CallKnown`/`Inst::CallUnknown` (we don't out-of-line the fields yet) and there are issues with the size of `Inst`, I'm inclined not to go down this path, at least at first. Can certainly look at this in a subsequent PR though!